### PR TITLE
Action items

### PR DESCRIPTION
**Pull request by Avogardo**

### CODE REVIEW

File: app/imports/api/actionItems/methods.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                'action-items-only-open-sprint-add',
+                'You can add new action item only in open sprints',
+            );
+        }
+
+        const projectId = sprint.projectId;
+
+        if (isProjectMember(projectId, userId)) {
+            const open = true;
+            const closeMessage = '';
```
Comment (mstanuch) [inline app/imports/api/actionItems/methods.js]: You can add default value for `open` and `closeMessage`.

File: app/imports/components/ActionItems/ActionItemsBoard/ActionItem.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        </CardActions>
+    </Card>
+);
+
+ActionItem.defaultProps = {
+    errorRemove: null,
+    idToRemove: '',
+    closeMessage: '',
+    assignee: {
+        name: 'Member',
```
Comment (mstanuch) [inline app/imports/components/ActionItems/ActionItemsBoard/ActionItem.jsx]: You should display that action item doesn't have assignee.

File: app/imports/components/ActionItems/ActionItemsBoard/ActionItems.jsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            startDate: PropTypes.instanceOf(Date).isRequired,
+            endDate: PropTypes.instanceOf(Date).isRequired,
+            assigneeId: PropTypes.string.isRequired,
+            assignee: PropTypes.shape({
+                name: PropTypes.string,
+                avatar: PropTypes.string,
+            }).isRequired,
+        }),
+    ).isRequired,
+    handlers: PropTypes.arrayOf(
```
Comment (mstanuch) [inline app/imports/components/ActionItems/ActionItemsBoard/ActionItems.jsx:228]: Separate Meteor from React.

File: app/imports/components/ActionItems/ActionItemsBoard/ActionItems.jsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    openToggleSnackbar: false,
+};
+
+ActionItems.propTypes = {
+    sprintId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    addActionItem: PropTypes.func.isRequired,
+    closeSnackBar: PropTypes.func.isRequired,
+    wrappedData: PropTypes.func.isRequired,
+    onData: PropTypes.func.isRequired,
```
Comment (mstanuch) [inline app/imports/components/ActionItems/ActionItemsBoard/ActionItems.jsx:210]: Separate react-komposer from react.

File: app/imports/components/ActionItems/ActionItemsBoard/ActionItemsToolbar.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import RenderCategories from './RenderCategories.jsx';
+
+const ActionItemsToolbar = ({
+    addActionItem,
+    isMember,
+    isClosed,
+    selectedState,
+    onChangeCategory,
+}) => {
+    const categories = [
```
Comment (mstanuch) [inline app/imports/components/ActionItems/ActionItemsBoard/ActionItemsToolbar.jsx]: Category isn't the best name. Maybe something with status?

File: app/imports/components/ActionItems/ActionItemsBoard/index.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    if (handlers.every(handler => handler.ready())) {
+        const userId = Meteor.userId();
+        const sprint = Sprints.findOne(sprintId);
+        const projectId = sprint.projectId;
+        const isMember = isProjectMember(projectId, userId);
+        const isModerator = isProjectModerator(projectId, userId);
+        const actionItems = ActionItemsCollection.find().fetch();
+
+        actionItems.map((actionItem) => {
+            const member = Meteor.users.find(actionItem.assigneeId).fetch();
```
Comment (mstanuch) [inline app/imports/components/ActionItems/ActionItemsBoard/index.js]: `findOne`?

File: app/imports/components/ActionItems/ActionItemsBoard/index.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            actionItem.assignee = {
+                name: member[0].profile.name,
+                avatar: member[0].profile.avatar,
+            };
+            return false;
+        });
+
+        onData(null, {
+            onData,
+            handlers,
```
Comment (mstanuch) [inline app/imports/components/ActionItems/ActionItemsBoard/index.js:86]: Separate Meteor from React.

File: app/imports/components/ActionItems/AddActionItem/CustomDatePicker.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react';
+import connectField from 'uniforms/connectField';
+
+import { DatePicker } from 'material-ui';
+
+
+const CustomDatePicker = props => (
+    <DatePicker
```
Comment (mstanuch) [inline app/imports/components/ActionItems/AddActionItem/CustomDatePicker.jsx]: It's too custom. At least allow to add other props.

**Review by mstanuch — CHANGES_REQUESTED** [review]
Also you have an bug - start date can be after deadline.

File: app/imports/components/ActionItems/ActionItemsBoard/ActionItems.jsx (hunk 3 above)
Comment (Avogardo) [reply app/imports/components/ActionItems/ActionItemsBoard/ActionItems.jsx:228]: Meteor and composer are necessary to properly working functions in index.js with onData wrapper. Cant figure better solution

**Review by mstanuch — APPROVED** [review]

**Pull request merged**